input/system/azure/logs.go

```diff
@@ -88,15 +92,18 @@ func getEventHubConsumerClient(config config.ServerConfig) (*azeventhubs.Consume
 			}
 		}
 		managedIdentityCredential, err := azidentity.NewManagedIdentityCredential(managedIdentityOptions)
-		if err != nil {
-			return nil, fmt.Errorf("failed to set up managed identity Azure credentials: %s", err)
+		if err == nil {
+			creds = append(creds, managedIdentityCredential)
+		} else {
+			errorMessages = append(errorMessages, "ManagedIdentityCredential: "+err.Error())
 		}
-		credential, err = azidentity.NewChainedTokenCredential([]azcore.TokenCredential{
-			workloadIdentityCredential,
-			managedIdentityCredential,
-		}, nil)
-		if err != nil {
-			return nil, fmt.Errorf("failed to use default Azure credentials: %s", err)
+		if len(creds) == 0 {
+			return nil, fmt.Errorf("failed to set up Azure credentials:\n\t" + strings.Join(errorMessages, "\n\t"))
```

Comment:
[note for reviewer] interestingly, even when I pass a bad client ID or not specifying client ID with multiple identities, it won't really fail here but will fail when actually trying to pull the log from the event hub. This is because this part is purely creating a cred, not really validating that cred. In other words, I don't know if this condition will ever meet, as managedIdentityCredential will always be created.
I think it's okay to leave it as is, but just wanted to leave a memo here.